### PR TITLE
Copy Accounting Year (and corresponding periods) to a new fiscal year

### PR DESCRIPTION
**Pull request by andersdd**
Db logic.  Also has qt-client changes required.

### CODE REVIEW

**Comment by shackbarth** [comment]
Is there an incident for this, and does it have anything to do with 
https://github.com/xtuple/xtuple/pull/2118
?

**Comment by andersdd** [comment]
No incident.  David B and others were debating this on Slack and I
already had most of the logic done from elsewhere so decided to put it
in place.

**Comment by ssshyam** [comment]
Unable to perform UAT: This PR depends on qt-client PR #576

File: foundation-database/public/functions/copyaccountingyearperiod.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+BEGIN
+
+  -- First create new fiscal year
+  INSERT INTO yearperiod (yearperiod_start, yearperiod_end, yearperiod_closed)
+    SELECT yearperiod_start + INTERVAL ' 1 year',  yearperiod_end + INTERVAL ' 1 year', FALSE
+      FROM yearperiod
+      WHERE yearperiod_id = pYearPeriodid
+  RETURNING yearperiod_id INTO _newYear;
+
+  IF (NOT _newYear > 0) THEN
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyaccountingyearperiod.sql]: If the insert fails then `_newYear` will be `null` and this `IF` will evaluate to false, so the exception won't be raised (welcome to NULL hell). It's safer to check `IF _newYear IS NULL`.

File: foundation-database/public/functions/copyaccountingyearperiod.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    RAISE EXCEPTION 'An error occurred creating the Fiscal Year';
+    RETURN -1;
```
Comment (gilmoskowitz) [inline foundation-database/public/functions/copyaccountingyearperiod.sql]: The `return` won't get executed. Better to include the `-1` in the message itself so the client can translate the message. See the [exception example](https://github.com/xtuple/qt-client/wiki/xTuple-Desktop-Client-Coding-Standards#plpgsql), at least until we redesign error handling again.

**Pull request merged**